internal/pkg/propagation/propagation.go

```diff
@@ -225,6 +221,10 @@ func (prop *Propagation) taintField(n ssa.Node, maxInstrReached map[*ssa.BasicBl
 }
 
 func (prop *Propagation) taintReferrers(n ssa.Node, maxInstrReached map[*ssa.BasicBlock]int, lastBlockVisited *ssa.BasicBlock) {
+	if !hasTaintableType(n) {
+		return
+	}
+
```

Comment:
What is the motivation of moving this block from `shouldNotTaint`?

Reply:
I swear I clicked on `Conversation` and it said `0 comments`.  Let me go read the PR description...

Reply:
Now that I'm caught up...

For clarity, we have an existing bug where a call that returns a boolean that should propagate taint between arguments would not have done so before this PR.  E.g. 
```golang
func foo(x, y interface[]) (ok bool) { 
  // ...
}

// ...
func bar(s Source) {
  var tainted interface{}
  if ok := foo(s, tainted) ; ok {
    sink(tainted) // TODO want
  }
}
```

If it's not too much of a hassle, I'd prefer to see a bug fix for this as a separate PR.  (Actually, I _really_ just want it visible as a separate commit in history, but since we squash rather than merge that necessitates a separate PR.)

Reply:
Thank you for pointing out this test case.

The move from `shouldNotTaint` to `taintReferrers` originally came from an existing failing test case involving the `copy` builtin. So to implement this change without failing tests, I would need to make the move first (justifying it with e.g. the test case you gave), and then make the change to only taint strings.

I can definitely do that, but I'm not sure I see the benefit (a slightly cleaner git history?), so to me this does feel like a hassle. If you feel strongly about it though, I'll do it.

I'm not sure that this requires a separate PR, as I could just undo the commit, add the boolean test case, commit the boolean test case + move in a first commit, then do the int change in a second commit, and finally force push. If that sounds gross to you though, we can go with the 2 PRs route.

Reply:
Done.